### DISCUSSION TITLE
Missing commands for kits

### DISCUSSION THREAD

**Discussion by jocoleman-sym** [microsoft/vscode-cmake-tools#3550] [Q&A] [answered]
I've installed the extension and I'm trying to set it up to work with an MSYS2 environment.

From the command pallet, I don't have any of the options relating to kits, like:
- Edit User-Local CMake Kits
- Scan for Kits
- Select a Kit

I'm able to run the commands if I add custom key bindings, but I can't see or run them from the command palette. What's going on?

**Comment by gcampbell-msft** [comment]
@symjocoleman Have you opened VS Code in a workspace that has a CMakeLists.txt? It could be that, if you've opened it in an empty folder, it won't activate the extension. 

Hopefully we can get you started!

**Comment by jocoleman-sym** [comment]
The workspace has a `CMakeLists.txt` file.

I tried another workspace, https://github.com/jameskbride/cmake-hello-world, and the kit's _are_ showing there.

It seems the issue is with the CMakePresets.json in my workspace, as if I copy that into the cmake-hello-world workspace and reload VS I lose the options for selecting a kit.

**Reply by gcampbell-msft** [reply, marked as answer]
Ahh, this is expected. Kits and CMakePreset.json are by design mutually exclusive. We suggest people to move to CMakePresets.json, but if you want to use kits, you cannot use CMakePresets.json
